feat(core): Prompt user to confirm password when changing email and mfa is disabled

konstantintieber opened the pull request:
## Summary

Changed the `PATCH /api/v1/me` endpoint to only allow changing the email of a user who does not have MFA enabled, when the correct `currentPassword` is also supplied in the request payload.

The personal settings page now displays a dialog upon clicking "Save" for users without MFA who changed their email, asking them to enter their current password to confirm the changes.

## Demo recording

https://www.loom.com/share/d53dd0fe7bef4bb684bac8a35a9b2af1?sid=0df5d116-812d-483c-a0b1-7d327fe073a6

## Related Linear tickets, Github issues, and Community forum posts

Closes PAY-3800


## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) <!--
   **Remember, the title automatically goes into the changelog.
   Use `(no-changelog)` otherwise.**
-->
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included. <!--
   A bug is not considered fixed, unless a test is added to prevent it from happening again.
   A feature is not complete without tests.
-->
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

konstantintieber committed:
feat(core): require password confirmation when changing user email

konstantintieber committed:
feat(core): Prompt user to enter password when changing email and mfa disabled

konstantintieber committed:
Remove unused import

konstantintieber committed:
Add bottom margin to the confirm password modal message

konstantintieber committed:
Add unit tests for new ConfirmPasswordModal

konstantintieber committed:
Fix unit tests

konstantintieber committed:
Ensure provided password for changing email is not logged

konstantintieber committed:
Update docs

konstantintieber committed:
Remove unused variable

konstantintieber committed:
Omit confirmation password from userService.update

konstantintieber committed:
Remove unneeded modal setup code

konstantintieber committed:
Rephrase error message for wrong password supplied to change email

Co-authored-by: Marc Littlemore <MarcL@users.noreply.github.com>

konstantintieber committed:
Fix formatting

MarcL reviewed (dismissed):
You'll need to fix the type linting issue but looks good otherwise.

konstantintieber committed:
Fix tests for MeController (three still failing)

konstantintieber committed:
Extract method for logic to verify user email change

konstantintieber commented on packages/cli/src/controllers/me.controller.ts:
```diff
@@ -103,10 +104,33 @@ export class MeController {
 
-		await this.externalHooks.run('user.profile.beforeUpdate', [userId, currentEmail, payload]);
+		if (!mfaEnabled && isEmailBeingChanged) {
+			// Shell users (users with null password) don't need to provide currentPassword
```
Nevermind, seems unreliable too. When debugging `req.user.authIdentities` for the user on my local instance, it's undefined.

konstantintieber committed:
Use type import when not used as value

konstantintieber committed:
Fix saml user integration test

- Avoid email change in this test, as that would require passing the currentPassword too.
- Removed non-existant password parameter

konstantintieber committed:
Update error message assertion

konstantintieber committed:
Throw ForbiddenError when email is changed but no password is set

konstantintieber committed:
Support "shell user" without password case

konstantintieber committed:
Remove unused import

konstantintieber committed:
Remove unused code in modal

konstantintieber committed:
Confirm current password in e2e test to change user email

konstantintieber committed:
Fix format

konstantintieber committed:
Fix e2e test

konstantintieber committed:
Fix format

konstantintieber committed:
Merge branch 'master' into pay-3800-permanent-account-takeover-via-unverified-email-change

konstantintieber reviewed (commented):
Trying to run e2e tests
